### PR TITLE
Fixes #6 - Add build instructions

### LINKED ISSUE #6: Add build instructions
1. How to build and test with Maven.
2. How to load into Eclipse and run and debug.

### PR DESCRIPTION
**Pull request by akerekes**

### CODE REVIEW

File: features/com.google.gcp.eclipse.suite.e45.feature/.settings/org.eclipse.core.resources.prefs [hunk 1]
```diff
@@ -0,0 +1,2 @@
+eclipse.preferences.version=1
+encoding/<project>=UTF-8
```
Comment (elharo) [inline features/com.google.gcp.eclipse.suite.e45.feature/.settings/org.eclipse.core.resources.prefs]: Is this correct syntax?

**Comment by elharo** [comment]
LGTM.

The travis failures are at head because I just turned it on, and are probably not related to this cl. I'm working on fixing those.

File: README.md [hunk 2]
```diff
@@ -1 +1,45 @@
...
+1. Setup JDK 7 in Eclipse
+  1. Select `Window/Preferences` (on Mac `Eclipse/Preferences`)
+  1. Under `Java/Installed JREs` click `Add`
+  1. Select Standard VM and click `Next`
+  1. Select the folder that contains the JDK 7 installation by clicking
+     `Directory`
+  1. Set the name to `jdk7`
+  1. Click `Finish`
+  1. Set the checkbox selected for the JDK you just added to mark it as default
+  1. Click `Apply`, then `OK`
```
Comment (briandealwis) [inline README.md]: We need a step to go to the _Execution Environments_ page and mark this VM as being compatible with _JavaSE-1.7_.

File: README.md [hunk 3]
```diff
@@ -1 +1,45 @@
...
+
+*** Steps to import
+
+1. Setup JDK 7 in Eclipse
+  1. Select `Window/Preferences` (on Mac `Eclipse/Preferences`)
+  1. Under `Java/Installed JREs` click `Add`
+  1. Select Standard VM and click `Next`
+  1. Select the folder that contains the JDK 7 installation by clicking
+     `Directory`
+  1. Set the name to `jdk7`
```
Comment (briandealwis) [inline README.md]: To be honest, I think it's better to name the VM with the vendor and build number. E.g., OpenJDK 1.7_b39. Really helpful if you hit something that seems to be a JIT or JVM bug.

File: README.md [hunk 4]
```diff
@@ -1 +1,45 @@
...
+
+* Development
+
+** Import into Eclipse
+
+*** Requirements
+
+1. Eclipse Mars or later
+
+1. The [m2e plugin](http://www.eclipse.org/m2e/) installed to import the project into Eclipse
```
Comment (briandealwis) [inline README.md]: Use _m2eclipse_, and note that it is often called _m2e_.

File: README.md [hunk 5]
```diff
@@ -1 +1,45 @@
...
+  1. Select `Window/Preferences` (on Mac `Eclipse/Preferences`)
+  1. Under `Java/Installed JREs` click `Add`
+  1. Select Standard VM and click `Next`
+  1. Select the folder that contains the JDK 7 installation by clicking
+     `Directory`
+  1. Set the name to `jdk7`
+  1. Click `Finish`
+  1. Set the checkbox selected for the JDK you just added to mark it as default
+  1. Click `Apply`, then `OK`
+1. Select `File/Import...` menu in Eclipse
```
Comment (briandealwis) [inline README.md]: We need a step to load the target file. Either by opening the `.target` file in the _Target Platform Editor_ (more direct), or by using _Preferences &rarr; Plug-in Development &rarr; Target Platforms_ and selecting the _Eclipse Mars 4.5.2_ target.

File: gcloud-eclipse-tools.launch [hunk 6]
```diff
@@ -15,12 +15,12 @@
 <booleanAttribute key="includeOptional" value="true"/>
 <stringAttribute key="location" value="${workspace_loc}/../runtime-gcloud-eclipse"/>
 <booleanAttribute key="org.eclipse.jdt.launching.ATTR_USE_START_ON_FIRST_THREAD" value="true"/>
-<stringAttribute key="org.eclipse.jdt.launching.JRE_CONTAINER" value="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/jdk7-google-v6-64"/>
```
Comment (briandealwis) [inline gcloud-eclipse-tools.launch]: This should actually use the Execution Environment with JavaSE-1.7.

File: features/com.google.gcp.eclipse.suite.e45.feature/.settings/org.eclipse.core.resources.prefs (hunk 1 above)
Comment (briandealwis) [inline features/com.google.gcp.eclipse.suite.e45.feature/.settings/org.eclipse.core.resources.prefs]: It does look strange, but yes.  You can change the encoding of particular files which would be recorded as `encoding//path/in/project=XXX`

File: gcloud-eclipse-tools.launch [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+<stringAttribute key="checked" value="[NONE]"/>
+<booleanAttribute key="clearConfig" value="false"/>
+<booleanAttribute key="clearws" value="false"/>
+<booleanAttribute key="clearwslog" value="false"/>
+<stringAttribute key="configLocation" value="${workspace_loc}/.metadata/.plugins/org.eclipse.pde.core/gcloud-eclipse-tools"/>
+<booleanAttribute key="default" value="true"/>
+<booleanAttribute key="includeOptional" value="true"/>
+<stringAttribute key="location" value="${workspace_loc}/../runtime-gcloud-eclipse"/>
+<booleanAttribute key="org.eclipse.jdt.launching.ATTR_USE_START_ON_FIRST_THREAD" value="true"/>
+<stringAttribute key="org.eclipse.jdt.launching.JRE_CONTAINER" value="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/jdk7"/>
```
Comment (briandealwis) [inline gcloud-eclipse-tools.launch]: I think this should specify an Execution Environment of JavaSE-1.7.

**Comment by briandealwis** [comment]
@akerekes You can apparently trigger a new build by [closing and re-open this PR](http://stackoverflow.com/a/17624403/600339).  That will cause the PR to be re-evaluated against master.

**Comment by akerekes** [comment]
The builds will fail until: 1. the travis branch is merged with master, the travis related changes cherrypicked into this branch. I suggest to skip this for now.

**Comment by akerekes** [comment]
@briandealwis PTAL

File: README.md [hunk 8]
```diff
@@ -48,33 +48,35 @@ This is an Eclipse plugin for building and deploying Google Cloud Platform appli
...
-1. Select `Existing Maven Projects` from the list.
+  1. Select `File/Import...` menu in Eclipse.
 
-1. Click `Browse...` to select the root directory and select the direcorty that
-   contains the project.
+  1. Select `Existing Maven Projects` from the list.
 
-1. Under `Projects:` the `pom.xml` files representing modules should be
-   displayed, make sure that all of them are selected, and click `Finish`.
+  1. Click `Browse...` to select the root directory and select the direcorty that
```
Comment (briandealwis) [inline README.md]: nit: directory

File: README.md [hunk 9]
```diff
@@ -1 +1,89 @@
...
+
+# Development
+
+## Import into Eclipse
+
+### Requirements
+
+1. Eclipse Mars or later
+
+1. The [m2eclipse plugin](http://www.eclipse.org/m2e/) (also called m2e) installed to import the project into Eclipse.
```
Comment (briandealwis) [inline README.md]: It seems a bit strange to talk about a singular _project_

File: README.md [hunk 10]
```diff
@@ -1 +1,89 @@
...
+     Environments:`.
+
+  1. The JDK just added should show up in the list on the right along with other
+     installed JDKs/JREs. Set the checkbox next the the JDK 7 added in the
+     previous steps to mark it as compatible with the `JavaSE-1.7` execution
+     environment.
+
+  1. Click `OK`.
+
+1. Import the project
```
Comment (briandealwis) [inline README.md]: It seems a bit strange to talk about a singular _project_

**Comment by briandealwis** [comment]
LGTM.  I don't feel strongly about project vs project*s*

File: .project [hunk 11]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>trunk</name>
```
Comment (elharo) [inline .project]: prefer 4 space indents.

File: README.md [hunk 12]
```diff
@@ -1 +1,89 @@
 This is an Eclipse plugin for building and deploying Google Cloud Platform applications.
+
+# Development
+
+## Import into Eclipse
+
+### Requirements
+
+1. Eclipse Mars or later
```
Comment (elharo) [inline README.md]: Mars --> 4.5

File: README.md [hunk 13]
```diff
@@ -1 +1,89 @@
...
+  1. Select `Existing Maven Projects` from the list.
+
+  1. Click `Browse...` to select the root directory and select the directory that
+     contains the project.
+
+  1. Under `Projects:` the `pom.xml` files representing modules should be
+     displayed, make sure that all of them are selected, and click `Finish`.
+
+  1. Maven will prompt you to install several additional plugin connector plugins from
+  [Tycho](https://eclipse.org/tycho/) if they are not already installed, click
```
Comment (elharo) [inline README.md]: run-on sentence. installed. Click

File: README.md [hunk 14]
```diff
@@ -1 +1,89 @@
...
+  `Finish` to install them. If further additional plugins are required as
```
Comment (elharo) [inline README.md]: If Eclipse prompts you to install additional plugins, do so.

File: README.md [hunk 15]
```diff
@@ -1 +1,89 @@
...
+  prompted by Eclipse, install them following the instructions.
+
+  1. After the installation is done, restart Eclipse as prompted.
```
Comment (elharo) [inline README.md]: Restart Eclipse when prompted.

File: eclipse/mars/.project [hunk 16]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>gcp-eclipse-mars</name>
```
Comment (elharo) [inline eclipse/mars/.project]: 4 space indents

File: plugins/com.google.gcp.eclipse.appengine.localserver/.project [hunk 17]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>com.google.gcp.eclipse.appengine.localserver</name>
```
Comment (elharo) [inline plugins/com.google.gcp.eclipse.appengine.localserver/.project]: 4 space indents

**Comment by elharo** [comment]
LGTM

File: .project (hunk 11 above)
Comment (akerekes) [inline .project]: This is a internal settings file used by Eclipse. I could not find the preferences to change its behavior.

File: eclipse/mars/.project (hunk 16 above)
Comment (akerekes) [inline eclipse/mars/.project]: internal settings, could not find way to customize it.

File: plugins/com.google.gcp.eclipse.appengine.localserver/.project (hunk 17 above)
Comment (akerekes) [inline plugins/com.google.gcp.eclipse.appengine.localserver/.project]: internal settings, could not find way to customize it.

**Pull request merged**